clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -109,8 +109,10 @@ public int throttleTimeMs() {
     @Override
     public Map<Errors, Integer> errorCounts() {
         Map<Errors, Integer> errorCounts = new HashMap<>();
-        data.topics().forEach(metadata ->
-            updateErrorCounts(errorCounts, Errors.forCode(metadata.errorCode())));
+        data.topics().forEach(metadata -> {
+            metadata.partitions().forEach(p -> updateErrorCounts(errorCounts, Errors.forCode(p.errorCode())));
```

Comment:
Why did we change this?

Reply:
It includes the partition-level error. It seems to me it follows the docs.
`
The number of each type of error in the response, including {@link Errors#NONE} and top-level errors as well as more specifically scoped errors (such as topic or partition-level errors).
`

Reply:
A metadata request has topics as the "scoped error". So, it looks to me that we introduced a regression here. We should check if there are similar cases for other requests.

Reply:
> A metadata request has topics as the "scoped error" (one cannot request a metadata request for a given partition).

I check the server-side code. You are right. This change should be reverted. I will file a PR to fix it ASAP. This behavior can be broken inadvertently (server-side/client-side add/remove specifically scoped error). not sure whether we can prevent it effectively (maybe add more tests)

Reply:
https://github.com/apache/kafka/pull/11127